lib/spack/spack/cmd/containerize.py

```diff
@@ -12,14 +12,41 @@
 level = "long"
 
 
+def setup_parser(subparser):
+    subparser.add_argument(
+        '-f', '--file', default=None,
+        dest='specfile', metavar='SPEC_YAML_FILE',
+        help="spack.yaml for which to generate the recipe. If not provided"
+             "try the env_dir provided by `spack -e`, if that also doesn't"
+             "exist try `./spack.yaml`.")
+    subparser.add_argument(
+        '-r', '--recipefile', default=None,
+        dest='recipefile',
+        help="name of the recipefile to generate. If None print to stdout.")
```

Comment:
Not having this was a deliberate design choice. In this PR we are maintaining more code to avoid a shell redirect, since:
```console
$ spack -D <spack-yaml-dir> containerize > <recipe-file>
```
has the same effect as:
```console
$ spack containerize --file=<spack-yaml-file> --recipefile=<recipe-file>
```
Also, in #14200 it was decided that container configuration should always reside in a `spack.yaml` file and not an arbitrarily named file (as it was in the original straw man implementation).

Reply:
That works for bash-scripted things but not as easily for python scripted versions. Yes I can probably come up with intercepting stdout, but that really sounds like an awful hack.

Reply:
@healther Don't need to. The command is based on APIs in the `spack.container` package:
```python
recipe = spack.container.recipe(config)
```
and you can use those directly.